File: Vector2D/Vector2D.cpp
#include "Vector2D.h"
#include <cmath>

// Реалізація конструктора за замовчуванням
Vector2D::Vector2D() : x(0.0), y(0.0) {}

// Реалізація конструктора з параметрами
Vector2D::Vector2D(double x, double y) : x(x), y(y) {}

// Реалізація конструктора копіювання
Vector2D::Vector2D(const Vector2D& other) : x(other.x), y(other.y) {}

// Реалізація деструктора
Vector2D::~Vector2D() {}

double Vector2D::getX() const { return x; }
double Vector2D::getY() const { return y; }

// Обчислення полярних координат
double Vector2D::getLength() const {
    return std::sqrt(x * x + y * y);
}

double Vector2D::getAngle() const {
    return std::atan2(y, x);
}

File: Vector2D_cs/Vector2D.cs
using System;

namespace VectorLibrary
{
    public class Vector2D
    {
        private double _x;
        private double _y;

        // Конструктор за замовчуванням
        public Vector2D()
        {
            _x = 0.0;
            _y = 0.0;
        }

        // Конструктор з параметрами
        public Vector2D(double x, double y)
        {
            _x = x;
            _y = y;
        }

        // Конструктор копіювання
        public Vector2D(Vector2D other)
        {
            _x = other._x;
            _y = other._y;
        }

        // Деструктор
        ~Vector2D()
        {
        }

        public double X { get { return _x; } }
        public double Y { get { return _y; } }

        public double GetLength()
        {
            return Math.Sqrt(_x * _x + _y * _y);
        }

        public double GetAngle()
        {
            return Math.Atan2(_y, _x);
        }
    }
}